lib/metadata/network_interface.go

```diff
@@ -54,7 +59,10 @@ type ContainerNetwork struct {
 	// The set of nameservers associated with this network - may be empty
 	Nameservers []net.IP `vic:"0.1" scope:"read-only" key:"dns"`
 
-	// The IP range for this network
-	FirstIP net.IP `vic:"0.1" scope:"read-only" key:"first_ip"`
-	LastIP  net.IP `vic:"0.1" scope:"read-only" key:"last_ip"`
+	// The IP ranges for this network
+	Pools []IPRange `vic:"0.1" scope:"read-only" key:"pools"`
+
+	// A reference to the vsphere network object
+	NetworkRef types.ManagedObjectReference `vic:"1.0" scope:"read-only" key:"network"`
+	Network    object.NetworkReference
```

Comment:
This, along with code in portlayer.go, I expect to see replaced with code from your PR.